.github/workflows/changelog-checker.yml

```diff
@@ -0,0 +1,47 @@
+# This workflow checks that there is either a 'pr/no-changelog' label applied to a PR
+# or there is a changelog/<pr number>.txt file associated with a PR for a changelog entry
+
+name: Check Changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to master
+    branches:
+      - master/*
+
+jobs:
+  # checks that a changelog entry is present for a PR
+  changelog-check:
+    # If there  a `pr/no-changelog` label we ignore this check
+    if: "!contains(github.event.pull_request.labels.*.name, 'pr/no-changelog')"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
+      - name: Check for changelog entry in diff
+        run: |
+          # check if there is a diff in the changelog directory
+          changelog_files=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/${{ github.event.pull_request.base.ref }}")" -- changelog/${{ github.event.pull_request.number }}.txt)
+
```

Comment:
Should it be an error to have more than one changelog file?  Similarly, do we want to try to catch an incorrectly numbered changelog file?

Reply:
I don't think it should be an error to have more than one. I tested this and it does wind up checking that the number matches - https://github.com/hashicorp/vault-plugin-mock/pull/76. You can see various permutations in the history on this PR.

Reply:
Yeah, what I meant was that it looks like this code wouldn't see a cl file for a different pr#, so we'd say "did not find a changelog entry" when we could give a more specific error like "incorrectly named changelog entry".

Reply:
I'll update the error to be more descriptive, also keeping in mind Calvin's thoughts in the above convo.